android/app/src/main/java/org/fossasia/openevent/fragments/FeedFragment.java

```diff
@@ -157,6 +166,24 @@ public void networkUnavailable() {
 
     }
 
+    private void showProgressBar(boolean show) {
+        if(show)
+            downloadProgressDialog.show();
+        else
+            downloadProgressDialog.dismiss();
+    }
+
+    private void setupProgressBar() {
+        downloadProgressDialog = new ProgressDialog(getContext());
+        downloadProgressDialog.setIndeterminate(true);
+        downloadProgressDialog.setProgressPercentFormat(null);
+        downloadProgressDialog.setProgressNumberFormat(null);
+        downloadProgressDialog.setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
+        downloadProgressDialog.setCancelable(false);
+        String shownMessage = String.format(getString(R.string.downloading_format), getString(R.string.menu_feed));
+        downloadProgressDialog.setMessage(shownMessage);
```

Comment:
Both scenarios are different as one is listening to progress and incrementing on changes and other one is intermediate. This one is not getting reused anywhere in the app, or else we would have created a single class. There is no code sharing between these use cases